snappy/snappy_compress.c

```diff
@@ -484,14 +524,20 @@ snappy_status snappy_compress_host(struct host_buffer_context *input, struct hos
 	return SNAPPY_OK;
 }
 
-snappy_status snappy_compress_dpu(struct host_buffer_context *input, struct host_buffer_context *output, uint32_t block_size, struct program_runtime *runtime)
+snappy_status snappy_compress_dpu(const unsigned char *in, size_t in_len, unsigned char *out, size_t *out_len,
+            void *wrkmem, struct program_runtime *runtime)
 {
+	// Set block size
+	uint32_t block_size = 4 * 1024;
```

Comment:
This should be a global variable, or better, a define (the value should not change during the program)